micrometer-tracing-reporters/micrometer-tracing-reporter-wavefront/src/main/java/io/micrometer/tracing/reporter/wavefront/WavefrontSpanHandler.java

```diff
@@ -354,10 +360,14 @@ private void send(TraceContext context, FinishedSpan span) {
 
     @Override
     public void run() {
-        while (!stop) {
+        while (!stop.get()) {
             try {
-                Pair<TraceContext, FinishedSpan> contextAndSpan = spanBuffer.take();
-                send(contextAndSpan._1, contextAndSpan._2);
+                SpanToSend spanToSend = spanBuffer.take();
+                if (spanToSend instanceof DeathPill) {
```

Comment:
What do you think about a "death pill" instance (could be an enum too) instead so that we can do this: `if (spanToSend == DEATH_PILL)`?

Reply:
Yep, good idea, will do this.